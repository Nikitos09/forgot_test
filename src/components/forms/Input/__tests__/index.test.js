import React from 'react';
import { normalize } from 'normalizr';
import { reduxForm } from 'redux-form';
import { Provider, connect } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Layout from './../index';

const form = reduxForm({
  form: 'test-form',
  touchOnChange: false
});

const mapStateToProps = () => ({});
const mapDispatchToProps = () => ({});
const Container = connect(mapStateToProps, mapDispatchToProps)(form(Layout));

const props = { name: 'test-field' };

const middlewares = [];
const mockStore = configureStore(middlewares);
let store;

describe(`redux-form Input`, () => {
  beforeEach(() => {
    store = mockStore({});
  });
  it('should match snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Container {...props} />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
