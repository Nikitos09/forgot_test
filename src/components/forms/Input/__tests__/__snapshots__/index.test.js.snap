// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux-form Input should match snapshot 1`] = `
<div
  className="form-group text-left"
>
  <div
    className="input-group"
  >
    <input
      anyTouched={false}
      array={
        Object {
          "insert": [Function],
          "move": [Function],
          "pop": [Function],
          "push": [Function],
          "remove": [Function],
          "removeAll": [Function],
          "shift": [Function],
          "splice": [Function],
          "swap": [Function],
          "unshift": [Function],
        }
      }
      asyncValidate={[Function]}
      autofill={[Function]}
      blur={[Function]}
      change={[Function]}
      className="form-control"
      clearAsyncError={[Function]}
      clearFields={[Function]}
      clearSubmit={[Function]}
      clearSubmitErrors={[Function]}
      destroy={[Function]}
      error={undefined}
      handleSubmit={[Function]}
      id={null}
      initialValues={undefined}
      initialize={[Function]}
      initialized={false}
      invalid={false}
      name="test-field"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      pure={true}
      reset={[Function]}
      resetSection={[Function]}
      submit={[Function]}
      submitSucceeded={false}
      touch={[Function]}
      triggerSubmit={undefined}
      type="text"
      untouch={[Function]}
      valid={true}
      value=""
      warning={undefined}
    />
  </div>
</div>
`;
